src/components/ThemedButton.tsx

```diff
@@ -33,15 +33,21 @@ const themeStyles = StyleSheet.create<{[x in ThemeVariant]: TextStyle}>({
 const ThemedButton = ({
   style = {},
   children,
+  onPress,
   ...rest
 }: TouchableOpacityProps & {
   children: React.ReactNode | React.ReactNode[];
+  onPress: () => void;
 }) => {
   const theme = useTheme();
+  const onButtonPress = useCallback(() => {
+    onPress;
+  }, [onPress]);
 
   return (
     <TouchableOpacity
       style={[style, styles.button, themeStyles[theme]]}
+      onPress={onButtonPress}
```

Comment:
You can use the `onPress` directly, the `useCallback` is normally used whenever you have dependencies. For instance, if you wanted to pass in the `ResourceType` or any other data. 

For context: `useCallback` memoizes the CB to prevent react from re-rendering the component. Whenever it can be avoided it's better since it'll have a minor memory hit :) (many minor hits add up though!)